plugins/services/src/js/reducers/serviceForm/Container.js

```diff
@@ -1,37 +1,191 @@
-import {SET} from '../../../../../../src/js/constants/TransactionTypes';
+import {SET, ADD_ITEM, REMOVE_ITEM} from '../../../../../../src/js/constants/TransactionTypes';
+import {combineReducers} from '../../../../../../src/js/utils/ReducerUtil';
+import {JSONReducer as volumes} from './Volumes';
 import VolumeConstants from '../../constants/VolumeConstants';
+import ValidatorUtil from '../../../../../../src/js/utils/ValidatorUtil';
 
-let {MESOS} = VolumeConstants.type;
+import docker from './Docker';
 
-function container(state = {}, {type, path, value}) {
-  let joinedPath = path && path.join('.');
-  let newState = Object.assign({}, state);
-  if (!newState.type) {
-    newState.type = MESOS;
+const {MESOS} = VolumeConstants.type;
+
+const containerJSONReducer = combineReducers({
+  type(state, {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    if (this.hasImage == null) {
+      this.hasImage = false;
+    }
+
+    if (this.noState == null) {
+      this.noState = true;
+    }
+
+    if (this.hasVolumes == null) {
+      this.hasVolumes = [];
+    }
+
+    const joinedPath = path.join('.');
+
+    if (joinedPath === 'container.docker.image') {
+      this.hasImage = !ValidatorUtil.isEmpty(value);
+    }
+
+    if (path[0] === 'localVolumes' && !this.hasImage) {
+      switch (type) {
+        case ADD_ITEM:
+          this.hasVolumes.push(true);
+          break;
+        case REMOVE_ITEM:
+          this.hasVolumes = this.hasVolumes.filter((item, index) => {
+            return index !== value;
+          });
+      }
+
+      // return this.hasVolumes.length !== 0 ? MESOS : state;
```

Comment:
I don't think this is needed, so I removed it, but @Poltergeist please review when you have a moment :)